impeller/toolkit/android/toolkit_android_unittests.cc

```diff
@@ -0,0 +1,63 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/fml/synchronization/waitable_event.h"
+#include "flutter/testing/testing.h"
+#include "impeller/toolkit/android/hardware_buffer.h"
+#include "impeller/toolkit/android/proc_table.h"
+#include "impeller/toolkit/android/surface_control.h"
+#include "impeller/toolkit/android/surface_transaction.h"
+
+namespace impeller::android::testing {
+
+class ToolkitAndroidTest : public ::testing::Test {
+ public:
+  void SetUp() override {
+    // The toolkit is only available on Android API levels over 29. Skip these
+    // tests everywhere else.
+    if (__builtin_available(android 29, *)) {
+    } else {
+      GTEST_SKIP() << "Platform too old for this test.";
```

Comment:
Given that we just went through a fire drill where tests weren't actually running (because they failed unsafely) - how can we guarantee that we get a failure somewhere if we accidentally stop running this on API 29+ ?

Reply:
We have the same problem with the existing ndk_helpers_unittests.

We do, in fact, currently run emulators that will run this test, but we should probably patch ... something to fail if this test never runs at all anywhere.

Reply:
Should I just do a GTEST_FAIL instead?

Reply:
GTEST_FAIL SGTM

Reply:
We have infra that runs an API 28 emulator. 

We should probably have a test that verifies we have sane/non-crashing behavior if you use this API below 29 though...

Reply:
> runs an API 28 emulator...

So, I can't make this a fail right?

> we have sane/non-crashing behavior if you use this API below 29 though

We are definitely missing a trick not using static analysis (like at-availability on iOS) for this. I spent a non-trivial amount of [time the other day wiring it up](https://discord.com/channels/608014603317936148/608021010377080866/1215804700046262334). But I haven't been able to get it to work reliably with our toolchain. Something to consider in addition to just checking on all emulator versions.